Behavioral.Automation.API/Bindings/HttpResponseSteps.cs

```diff
@@ -0,0 +1,289 @@
+﻿using System.Text.RegularExpressions;
+using Behavioral.Automation.API.Context;
+using Behavioral.Automation.API.Models;
+using Behavioral.Automation.API.Services;
+using Behavioral.Automation.Configs.utils;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+using Polly;
+using TechTalk.SpecFlow;
+
+namespace Behavioral.Automation.API.Bindings;
+
+[Binding]
+public class HttpResponseSteps
+{
+    private readonly ApiContext _apiContext;
+    private readonly HttpService _httpService;
+    private readonly int _retryCount = 10;
+    private readonly TimeSpan _retryDelay = TimeSpan.FromSeconds(3);
+
+    public HttpResponseSteps(ApiContext apiContext, HttpService httpService)
+    {
+        _apiContext = apiContext;
+        _httpService = httpService;
+    }
+
+    [Then("response attachment filename is \"(.*)\"")]
+    public void ThenResponseAttachmentFilenameIs(string filename)
+    {
+        if (_apiContext.Response is null) throw new NullReferenceException("Http response is empty.");
+        var contentDispositionHeader = _apiContext.Response.Content.Headers.ContentDisposition;
+        if (contentDispositionHeader == null)
+        {
+            Assert.Fail("Response header \"ContentDisposition disposition\" is null");
+        }
+        if (!contentDispositionHeader.ToString().StartsWith("attachment"))
+        {
+            Assert.Fail("Content disposition is not attachment?");
+        }
+
+        if (!contentDispositionHeader.FileName.Equals(filename))
+        {
+            Assert.Fail($"filename is wrong.\n\nActual result: {contentDispositionHeader.FileName}\nExpected result: {filename}");
+        }
+    }
+
+    [Given("response attachment is saved as \"(.*)\"")]
+    public void GivenResponseAttachmentIsSavedAs(string filePath)
+    {
+        if (_apiContext.Response is null) throw new NullReferenceException("Http response is empty.");
+        var fullPath = Path.GetFullPath(Path.Join(Directory.GetCurrentDirectory(), filePath))
+            .NormalizePathAccordingOs();
+
+        var directoryPath = Path.GetDirectoryName(fullPath);
```

Comment:
Looks like yes. The logic is to check whether a directory exists and create if not.

Reply:
It is not clear here what could be written in filePath - full path or relative path? Because of that further manipulations with this value are hard to understand

Reply:
Btw, 
![image](https://user-images.githubusercontent.com/88668184/188095288-81a4c161-54eb-4a4f-b891-83ba52092dad.png)

Reply:
Fixed